index.js

```diff
@@ -1,141 +1,77 @@
-var convert   = require('convert-source-map');
-
-var PREFIX = "var jade = require('jade/lib/runtime.js');\nmodule.exports=";
+'use strict'
+
+const pug = require('pug')
+const xtend = require('xtend')
+const through = require('through')
+const convert = require('convert-source-map')
+const tools = require('browserify-transform-tools')
+const SourceMapGenerator = require('source-map').SourceMapGenerator
+
+const defaultOptions = {
+  pretty: false,
+  compileDebug: true
+}
 
-var defaultJadeOptions = {
-  path: __dirname,
-  compileDebug: true,
-  pretty: true,
-};
+// RegEx
+const FILE_EXTENSION_REGEX = /\.(pug|jade)$/
+const LINE_NUMBER_REGEX = /^;jade_debug_line = ([0-9]+)/
 
-function getTransformFn(options) {
-  var key;
-  var opts = {};
-  for(key in defaultJadeOptions) {
-    opts[key] = defaultJadeOptions[key];
-  }
+function generateSourceMaps (src, compiled, file) {
+  const lines = compiled.split('\n')
+  const generator = new SourceMapGenerator({ file: `${file}.js` })
 
-  options = options || {};
-  for(key in options) {
-    opts[key] = options[key];
-  }
+  lines.forEach((line, num) => {
+    const match = line.match(LINE_NUMBER_REGEX)
+    if (!match || match[1] < 1) return
 
-  return function (file) {
-    if (!/\.(pug|jade)$/.test(file)) return through();
+    generator.addMapping({
+      source: file,
+      generated: { column: 0, line: num + 2 },
+      original: { column: 0, line: Number(match[1]) }
+    })
+  })
 
-    var data = '';
-    return through(write, end);
+  generator.setSourceContent(file, src)
 
-    function write (buf) {
-      data += buf;
-    }
-    function end () {
-      var _this = this;
-      configData = transformTools.loadTransformConfig('browserify-jade', file, function(err, configData) {
-        if(configData) {
-          var config = configData.config || {};
-          for(key in config) {
-            opts[key] = config[key];
-          }
-        }
+  const map = convert.fromJSON(generator.toString())
+  lines.push(map.toComment())
 
-        var result = compile(file, data, opts);
-        result.dependencies.forEach(function(dep) {
-          _this.emit('file', dep);
-        });
-        _this.queue(result.body);
-        _this.queue(null);
-      });
-    }
-  };
+  return lines.join('\n')
 }
 
-module.exports = getTransformFn();
-module.exports.jade = getTransformFn;
-module.exports.root = null;
-module.exports.register = register;
+function pugify (file, opts) {
+  if (!FILE_EXTENSION_REGEX.test(file)) return through()
 
-function register() {
-  require.extensions['.pug', '.jade'] = function(module, filename) {
-    var result = compile(filename, fs.readFileSync(filename, 'utf-8'), {compileDebug: true});
-    return module._compile(result.body, filename);
-  }
-}
+  let buffer = ''
 
-function replaceMatchWith(match, newContent)
-{
-  var src = match.input;
-  return src.slice(0, match.index) + newContent + src.slice(match.index + match[0].length);
-}
+  return through(
```

Comment:
`browserify-transform-tools` has a `transformTools.makeStringTransform` method. Maybe switch to that?